pkg/telemetry/phonehome/gatherer_test.go

```diff
@@ -30,30 +32,91 @@ func (s *gathererTestSuite) TestNilGatherer() {
 
 func (s *gathererTestSuite) TestGatherer() {
 	t := mocks.NewMockTelemeter(gomock.NewController(s.T()))
+	g := newGatherer("Test", t, 24*time.Hour)
 
 	t.EXPECT().Track("Updated Test Identity", nil,
-		matchOptions(telemeter.WithTraits(map[string]any{"key": "value"}))).Times(1)
+		matchOptions(telemeter.WithTraits(map[string]any{"key": "value"}))).
+		Times(1).Do(func(any, any, ...any) { g.Stop() })
 	t.EXPECT().Track("Test Heartbeat", nil,
-		matchOptions(telemeter.WithTraits(nil))).Times(1)
+		matchOptions(telemeter.WithTraits(nil))).
+		Times(1).Do(func(any, any, ...any) { g.Stop() })
 
 	props := make(map[string]any)
-	var i int64
+	var i atomic.Int64
 
-	g := newGatherer("Test", t, 24*time.Hour)
 	g.AddGatherer(func(context.Context) (map[string]any, error) {
-		atomic.AddInt64(&i, 1)
+		i.Add(1)
 		props["key"] = "value"
-		g.Stop()
 		return props, nil
 	})
 	g.Start()
 	<-g.ctx.Done()
-	g.procs.Wait()
 	s.Equal("value", props["key"], "the gathering function should have been called")
-	s.Equal(int64(1), i)
+	s.Equal(int64(1), i.Load())
 	g.Start()
 	<-g.ctx.Done()
-	g.procs.Wait()
 	s.Equal("value", props["key"], "the gathering function should have been called")
-	s.Equal(int64(2), i)
+	s.Equal(int64(2), i.Load())
+}
+
+func (s *gathererTestSuite) TestGathererTicker() {
```

Comment:
It seems like this now has some flakes if ran multiple times:
```
-test.v -test.paniconexit0 -test.run ^\QTestConfig\E$/^\QTestGathererTicker\E$ -testify.m ^TestGathererTicker$ -test.count=100

=== RUN   TestConfig/TestGathererTicker
    controller.go:98: missing call(s) to *mocks.MockTelemeter.Track(is equal to Test Heartbeat (string), is nil, &{    map[] map[]}) /Users/dhaus/go/src/github.com/stackrox/rox/pkg/telemetry/phonehome/gatherer_test.go:73
    controller.go:98: aborting test due to missing call(s)
    --- FAIL: TestConfig/TestGathererTicker (0.00s)
```

Reply:
The `lastTrack` signal wasn't constructed with `NewSignal()`.